### ISSUE TITLE
Computation time needed for HalfCheetah

### ISSUE THREAD

**Issue by donamin** [google-research/batch-ppo#8]
Hi
I downloaded the repository and tried to replicate the paper results in Cheetah environment.
But the code seems to be super slow. As you can see in the picture, reward improvement starts after 4 hours and it gets to the early peak after 8 hours (which according to the paper, should happen in less than an hour). Am I doing something wrong?

![screenshot from 2017-09-28 09-15-01](https://user-images.githubusercontent.com/4231500/30952061-cc6a2358-a42e-11e7-8dba-33910f47d4bb.png)

I have Linux Mint and I installed tensorflow GPU version (however the use_gpu flag is false in the code by default). This is my system config:
Hexa core Intel Core i7-4930K (-HT-MCP-) cache: 12288 KB
GeForce GTX 1060 6GB/PCIe/SSE2

**Comment by danijar** [comment]
The performance curve is consistent with the paper in number of training steps, but running much slower for you. Here are a few things you can check:

- Can you verify that this is only running on CPU? For example check whether `nvidia-smi` shows any GPU memory usage. For such small networks running on CPU is quite a bit faster since transferring the data to GPU takes longer than the actual network computation.

- Please also measure the performance of the environment under your setup:
    ```shell
    python -c "import gym,time;d=10000;e=gym.make('HalfCheetah-v1');s=time.time();e.reset();[e.reset() if e.step(e.action_space.sample())[2] else 0 for _ in range(d)];print(d/(time.time()-s),'FPS')"
    ```

- Is TensorFlow compiled using all available optimizations, especially the full instruction set? If not, there will be warnings along the lines of `The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.` You might have to compile TensorFlow from source for this.

- The `num_agents` value in the config determines how many environments to run in parallel. It's an important setting for best performance. Could you set `num_agents = 25` and see if this improves performance for you?

I just re-ran the current repository on a 6 core CPU (Intel(R) Xeon(R) CPU E5-1650 v4 @ 3.60GHz). This is a faster CPU than we used for the graphs in the paper and I had some background tasks running, but the results are faster than what we reported in the paper:

![agents-performance-cheetah](https://user-images.githubusercontent.com/2111293/31184740-2c1eb352-a922-11e7-88dd-f2f622127421.png)

**Comment by danijar** [comment]
Closed due to inactivity and because we cannot reproduce the problem. Please comment again on the issue if you tried these steps and the problem remains.
